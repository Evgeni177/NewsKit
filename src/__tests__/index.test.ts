import * as newskit from '..';

describe('index exports', () => {
  test('all expected functions and components are exported', () => {
    expect(
      // Exclude icons as these will change between local and CI builds.
      Object.keys(newskit).filter(
        exportKey => !exportKey.match(/^Icon(?!Button)/),
      ),
    ).toMatchSnapshot();
  });

  test('no styled components are exported', () => {
    expect(
      // We should be prefixing all our internal styled comps with Styled - and these should not be exported.
      Object.keys(newskit).filter(exportKey => exportKey.startsWith('Styled')),
    ).toEqual([]);
  });
});
