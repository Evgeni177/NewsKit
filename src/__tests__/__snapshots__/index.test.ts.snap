// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index exports all expected functions and components are exported 1`] = `
Array [
  "AspectRatio",
  "AudioPlayer",
  "PopoutButton",
  "Byline",
  "Block",
  "Button",
  "ButtonSize",
  "isButtonLink",
  "Caption",
  "CaptionInset",
  "Card",
  "CardInset",
  "Consent",
  "ConsentSettingsLink",
  "DateTime",
  "Divider",
  "Drawer",
  "ExperimentationWeb",
  "Flag",
  "FlagSize",
  "Form",
  "Grid",
  "Cell",
  "Visible",
  "Hidden",
  "Headline",
  "IconButton",
  "IndeterminateProgressIndicator",
  "Svg",
  "toNewsKitIcon",
  "customToNewsKitIcon",
  "useClientSide",
  "Image",
  "instrumentationHandlers",
  "composeInstrumentationMiddleware",
  "instrumentationMiddleware",
  "InstrumentationProvider",
  "withInstrumentation",
  "useInstrumentation",
  "mergeContexts",
  "createEventInstrumentation",
  "EventTrigger",
  "Link",
  "A",
  "LinkStandalone",
  "OrderedList",
  "UnorderedList",
  "ScreenReaderOnly",
  "Scroll",
  "ShareBar",
  "Slider",
  "StatefulSlider",
  "LabelPosition",
  "StackChild",
  "AlignSelfValues",
  "Stack",
  "Flow",
  "StackDistribution",
  "Standfirst",
  "Tab",
  "Tabs",
  "TabSize",
  "TabsDistribution",
  "Tag",
  "TagSize",
  "Tealium",
  "TextBlock",
  "TextInput",
  "TextInputSize",
  "compileTheme",
  "createTheme",
  "withTheme",
  "useTheme",
  "ThemeProvider",
  "Global",
  "newskitDarkTheme",
  "newskitLightTheme",
  "Devices",
  "TitleBar",
  "Heading1",
  "H1",
  "Heading2",
  "H2",
  "Heading3",
  "H3",
  "Heading4",
  "H4",
  "Heading5",
  "H5",
  "Heading6",
  "H6",
  "ParagraphText",
  "ParagraphDropCap",
  "Paragraph",
  "P",
  "Subscript",
  "Sub",
  "Superscript",
  "Sup",
  "getDeviceQueryFromTheme",
  "getAspectRatioStyles",
  "getSSRId",
  "getMediaQuery",
  "getMediaQueries",
  "getMediaQueryFromTheme",
  "isResponsive",
  "getDefaultedValue",
  "getValueFromTheme",
  "getResponsiveValueFromTheme",
  "css",
  "styled",
  "getTypographyPresetFromTheme",
  "getMotionFromTheme",
  "getFontsFromTheme",
  "getColorFromTheme",
  "getSizingFromTheme",
  "getSpacingFromTheme",
  "getSpacingInsetFromTheme",
  "getMarginPresetFromTheme",
  "getBorderFromTheme",
  "getShadowFromTheme",
  "getFontSizingFromTheme",
  "getTypographyPreset",
  "getSpacingInlineHorizontal",
  "getSpacingInlineVertical",
  "getSpacingStackHorizontal",
  "getSpacingStackVertical",
  "getSpacingInset",
  "getSizing",
  "getMinHeight",
  "getMinWidth",
  "getHeight",
  "getWidth",
  "getSize",
  "getMaxSize",
  "getMinSize",
  "getSpace",
  "getWeight",
  "getPresetStyles",
  "getStylePresetFromTheme",
  "getStylePreset",
  "getSingleStylePreset",
  "deepMerge",
  "withDefaultProps",
  "VolumeControl",
]
`;
